src/test/java/org/apache/commons/beanutils2/collections/AbstractObjectTest.java

```diff
@@ -0,0 +1,338 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.beanutils2.collections;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.io.Serializable;
+
+/**
+ * Abstract test class for {@link java.lang.Object} methods and contracts.
+ * <p>
+ * To use, simply extend this class, and implement
+ * the {@link #makeObject()} method.
+ * <p>
+ * If your {@link Object} fails one of these tests by design,
+ * you may still use this base set of cases.  Simply override the
+ * test case (method) your {@link Object} fails.
+ * <p>
+ * Copied from Commons Collection 4.
+ * @see <a href="https://github.com/apache/commons-collections/blob/master/src/test/java/org/apache/commons/collections4/AbstractObjectTest.java">AbstractObjectTest.java</a>
```

Comment:
Why is this copied when we depend on Commons Collection 4?

Reply:
If I remember its a test only `/src/test/java` class in Commons Collections so it is not in the JAR file.

Reply:
@melloware 
Then just add a test dependency on the Commons Collection test jar ;-)

Reply:
I would not mention it if it didn't exist... https://repo1.maven.org/maven2/org/apache/commons/commons-collections4/4.4/

Reply:
OHHHH.  I see its actually a `<classifier>test</classifier>`  I don't love when libraries do this because its not advertised nor does it show up on Maven Central Searches!  Yeah @landsman most of the classes can be removed and that added to pom.xml now.

Reply:
Of course, it shows up: https://search.maven.org/search?q=fc:org.apache.commons.collections4.AbstractObjectTest